crates/nu-command/src/filters/find.rs

```diff
@@ -320,6 +320,29 @@ fn find_with_rest_and_highlight(
     };
     let ls_colors = get_ls_colors(ls_colors_env_str);
 
+    let input = if let PipelineData::Value(Value::String { ref val, span }, _) = input {
+        if val.contains('\n') {
+            Value::List {
+                vals: {
+                    let split_char = if val.contains("\r\n") { "\r\n" } else { "\n" };
+                    val.split(split_char)
```

Comment:
I think this is better expressed by just using `val.lines()` as this would tolerate mixed line endings as well and is more terse and explicit.

(I know we use this pattern in a few places like the handling of the external stream below but I am not sure if we gain anything major from it)